src/ansys/dpf/composites/ply_wise_data.py

```diff
@@ -0,0 +1,87 @@
+"""Methods to get ply wise data from a result field."""
+
+from enum import Enum, IntEnum
+from typing import Union
+
+from ansys.dpf.core import Field, MeshedRegion, Operator, operators
+from ansys.dpf.gate.common import locations
+
+__all__ = ("ReductionStrategy", "get_ply_wise_data")
+
+
+class ReductionStrategy(Enum):
+    """The reduction strategy to get from spot values (BOT, MID, TOP) to a single value."""
+
+    MIN = "MIN"
+    MAX = "MAX"
+    AVG = "AVG"
+    BOT = "BOT"
+    MID = "MID"
+    TOP = "TOP"
+
+
+def get_ply_wise_data(
+    field: Field,
+    ply_name: str,
+    mesh: MeshedRegion,
+    reduction_strategy: ReductionStrategy = ReductionStrategy.AVG,
+    requested_location: str = locations.elemental_nodal,
+    component: Union[IntEnum, int] = 0,
+) -> Field:
+    """Get ply-wise data from a field.
+
+    Parameters
+    ----------
+    field:
+        The field to extract data from.
+    ply_name:
+        The name of the ply to extract data from.
+    mesh:
+        The meshed region. Needs to be enriched with composite information.
+        Use CompositeModel.get_mesh() to get the meshed region.
+    reduction_strategy:
+        The reduction strategy to get from spot values (BOT, MID, TOP) to a single value
+        per corner node and layer. The default is AVG.
+    requested_location:
+        The location of the output field. The default is elemental_nodal. Supported are
+        elemental_nodal, elemental, and nodal.
+    component:
+        The component to extract data from. Can be an int or an IntEnum. The default is 0.
```

Comment:
```suggestion
        Name of the ply to extract data from.
    mesh :
        Meshed region. Needs to be enriched with composite information.
        Use the ``CompositeModel.get_mesh()`` method to get the meshed region.
    reduction_strategy :
        Reduction strategy for getting from spot values (BOT, MID, TOP) to a single value
        per corner node and layer. The default is ``AVG``.
    requested_location :
        Location of the output field. The default is ``"elemental_nodal"``. Options are
        ``"elemental"``, ``"elemental_nodal"``, and ``"nodal"``.
    component : int or IntEnum, optional
        Component to extract data from. The default is ``0``.
```
List default first and then provide a list of all options in alphabetical order after.